### DISCUSSION TITLE
Woche 10 - Projekt Zwischenstand

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#38] [Student Exchange]
Hallo zusammen,
zum Abschluss des dritten Semesters wollen wir euch durch alle von uns erarbeiteten 📄 Dokumente führen.

Zu Beginn haben wir alle relevanten Anforderungen an unsere Software in dem sogenanntem [SRS](https://github.com/Christian-2003/CrInGE/blob/master/docs/SoftwareAnforderungsspezifikation.md) festgehalten.
Hier findet ihr unter anderem auch die 💭 Vision unseres Projektes.

Anschließend haben wir in mehreren Schritten die [architektonisch relevanten Anforderungen](https://github.com/Christian-2003/CrInGE/blob/master/docs/ArchitektonischRelevanteAnforderungen.md) immer weiter spezifiziert und konkretisiert. Diese 🏗️ Architekturentscheidungen haben wir final nach dem arc42 Muster festgehalten.
Anzumerken ist an dieser Stelle, dass wir bereits ein wenig für kommendes Semester vor gedacht haben und sämtliche 12 Kapitel behandelt haben.

Unseren gesamten 💿 Quellcode könnt ihr unter dem [src Verzeichnis](https://github.com/Christian-2003/CrInGE/tree/master/src) in diesem Repository unter Code finden.
Des weitern könnt Ihr unsere [💼  Projektmanagement](https://github.com/users/Christian-2003/projects/2), welches mittels GitHub Projects umgesetzt ist, auch in diesem Repository unter Projects finden.
In diesem Projekt findet Ihr unter anderem eine [Liste aller 👥 User Stories](https://github.com/users/Christian-2003/projects/2/views/5), als auch eine [Liste aller 📑 Aufgaben](https://github.com/users/Christian-2003/projects/2/views/1). 
Außerdem könnt ihr euch eine [📅 Roadmap](https://github.com/users/Christian-2003/projects/2/views/2), welche den zeitlichen Zusammenhang zwischen den absolvierten Aufgaben zeigt, anschauen, auf einem [📋 Board](https://github.com/users/Christian-2003/projects/2/views/3) einen Überblick über die Aufgaben der jeweiligen ✔️ Status bekommen oder [📈 Statistiken](https://github.com/users/Christian-2003/projects/2/insights) zu unserem Projekt einsehen.
In dem kompletten Projekt könnt ihr jeweils nach Schlüsselwörtern 🔍 filtern, wie zum Beispiel: "iteration:Woche 2 & 3".

Die [🖥️ Präsentation](https://github.com/Christian-2003/CrInGE/blob/master/docs/Halbzeitpr%C3%A4sentation/Pr%C3%A4sentation.pdf) und das [🧾 Handout](https://github.com/Christian-2003/CrInGE/blob/master/docs/Halbzeitpr%C3%A4sentation/Handout.pdf) zur Halbzeit sind ebenfalls innerhalb des docs Ordner zu finden.

###### Demo Screenshots
Hier könnt Ihr einen Gesamtüberblick über den momentanen Editor bekommen:
![Editor gesamt](https://github.com/Christian-2003/CrInGE/assets/136824971/ece1263b-5aaf-45c8-94fe-233bacd78117)
Wie ihr seht habt Ihr an der rechten Seite Bilder, welche ihr auf das Raster der Map hinzufügen könnt:
![Map Raster](https://github.com/Christian-2003/CrInGE/assets/136824971/692a2ca9-2b04-4fbd-98cb-3533ddf84d17) ![Bilder Vorlagen](https://github.com/Christian-2003/CrInGE/assets/136824971/e3d8833f-d9a6-4fec-af3e-611e93cc0262)
Zu guter Letzt habt Ihr am oberen Bildschirmrand die Möglichkeit, zwischen verschiedenen Bearbeitungsmodi zu wählen:
![image](https://github.com/Christian-2003/CrInGE/assets/136824971/610cf450-4a19-492e-80b9-3b78f66c3b2e)

###### Engine Screenshots
Hier könnt ihr den momentanen Gesamtüberblick über die Engine bekommen:
![Screenshot 2023-12-22 083250](https://github.com/Christian-2003/CrInGE/assets/80711177/d3cb76f3-fb1d-4c8f-9017-f62c6729ec37)
Hier kann man eine gerenderte Videospielwelt sehen.

![Screenshot 2023-12-22 083311](https://github.com/Christian-2003/CrInGE/assets/80711177/3d8cf599-e0c1-41fc-97ff-9d7ff186f390)
Zu Debugging-Zwecken kann man weitere Debugging-Informationen (wie bspw. Hitboxen und Chunkgrenzen) anzeigen lassen.

Somit sind wir auch schon am Ende der Übersicht angelangt.
Wir freuen uns auf euer Feedback und werden im nächsten Semester wieder aktiv werden und von unserem Fortschritt berichten.

**Comment by Bjoern04** [comment]
Das sieht wirklich gut aus! Eine ausgezeichnete Übersicht über alle Dokumente. In Bezug auf eure SRS hätte ich eine zusätzliche Anmerkung. Bei den nicht-funktionalen Anforderungen wäre es hilfreich, genauere Messwerte zur Überprüfung der individuellen Ziele hinzuzufügen. Ähnlich wie bei der Leistungsanforderung mit dem Ziel von mindestens 20 FPS und der Begrenzung der Arbeitsspeicherbelastung auf maximal 4 GB – ihr könntet zusätzlich einen Maximalwert für die Ladezeiten und einen Mindestwert für die Zuverlässigkeit in Prozent festlegen.
